lamb/src/main/java/com/scalar/kelpie/lamb/ArgumentBuilder.java

```diff
@@ -0,0 +1,331 @@
+package com.scalar.kelpie.lamb;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.concurrent.ThreadSafe;
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonNumber;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+
+@ThreadSafe
+public class ArgumentBuilder {
+  private final JsonObject baseArguments;
+  private final JsonObject variableConfig;
+  private final Map<String, AtomicInteger> stateCounters = new ConcurrentHashMap<>();
+
+  private static final String TYPE = "type";
+  private static final String PATTERN = "pattern";
+  private static final String LIST = "list";
+  private static final String LENGTH = "length";
+  private static final String START = "start";
+  private static final String END = "end";
+
+  private static final String SEQUENTIAL_PATTERN = "SEQUENTIAL";
+  private static final String RANDOM_PATTERN = "RANDOM";
+  private static final String TIMESTAMP_PATTERN = "TIMESTAMP";
+
+  private static final String CHARACTERS = "abcdefghijklmnopqrstuvwxyz0123456789";
+
+  public ArgumentBuilder(JsonObject baseArguments, JsonObject variableConfig) {
+    this.baseArguments = baseArguments;
+    this.variableConfig = variableConfig;
+  }
+
+  public JsonObject build() {
+    JsonObjectBuilder builder = Json.createObjectBuilder();
+
+    for (Map.Entry<String, JsonValue> entry : baseArguments.entrySet()) {
+      JsonValue value = getJsonValue(entry.getValue());
+      builder.add(entry.getKey(), value);
+    }
+
+    return builder.build();
+  }
+
+  private JsonValue getJsonValue(JsonValue baseValue) {
+    JsonValue.ValueType type = baseValue.getValueType();
+
+    if (type == JsonValue.ValueType.ARRAY) {
```

Comment:
Should it be like this?
```suggestion
    if (type.equals(JsonValue.ValueType.ARRAY)) {
```

Reply:
I think it's OK since JsonValue.ValueType is enum.